lib/jsdom/living/nodes/HTMLElement-impl.js

```diff
@@ -48,12 +54,25 @@ class HTMLElementImpl extends ElementImpl {
     // Not completely spec compliant due to e.g. incomplete implementations of disabled for form controls, or no
     // implementation at all of isTrusted.
 
-    if (this.disabled) {
+    if (this._clickInProgress) {
+      return false;
+    }
+
+    this._clickInProgress = true;
+
+    if (this.hasAttribute("disabled")) {
       return false;
     }
 
     const event = MouseEvent.createImpl(["click", { bubbles: true, cancelable: true }], {});
+
+    // Run synthetic click activation steps. According to the spec,
+    // this should not be calling dispatchEvent, but it matches browser behavior.
+    // See: https://www.w3.org/Bugs/Public/show_bug.cgi?id=12230
```

Comment:
Maybe also https://github.com/whatwg/html/issues/805. I guess I did never link to that here, hrm.